pkg/ccl/streamingccl/streamproducer/event_stream.go

```diff
@@ -50,6 +50,7 @@ type eventStream struct {
 	// Fields below initialized when Start called.
 	rf          *rangefeed.RangeFeed          // Currently running rangefeed.
 	streamGroup ctxgroup.Group                // Context group controlling stream execution.
+	closeChan   chan struct{}                 // Channel signaled to close the stream loop.
```

Comment:
I think it would be nice to keep 1 channel.  

If I understood the problem you're describing, then closeChan
could be renamed to doneChan.  errCh can be removed.  When error happens (maybeSetError), close doneChan, and save the terminal error.